crates/chainspec/src/spec.rs

```diff
@@ -637,22 +637,33 @@ impl ChainSpec {
 
 impl From<Genesis> for ChainSpec {
     fn from(genesis: Genesis) -> Self {
+        let mut chain_spec = ChainSpec::from(genesis.config.clone());
+        chain_spec.genesis_header =
+            SealedHeader::new_unhashed(make_genesis_header(&genesis, &chain_spec.hardforks));
+        chain_spec.genesis = genesis;
+
+        chain_spec
+    }
```

Comment:
Main change was moving the config related components from here into the new From<ChainConfig> impl